packages/server/src/unstable-core-do-not-import/parser.ts

```diff
@@ -110,5 +120,16 @@ export function getParseFn<TType>(procedureParser: Parser): ParseFn<TType> {
     };
   }
 
+  if ('~standard' in parser) {
+    // StandardSchemaEsque
+    return async (value) => {
+      const result = await parser['~validate']({ value });
+      if (result.issues) {
+        throw new SchemaError(result.issues);
+      }
```

Comment:
If `result.reason` was an `Error`-type, we could've just rethrown here

But for now I think we gotta do something like

```ts
if (result.issues) {
  throw new Error("Validation error", { cause: result.issues })
}
```

For us, it would be a bit simpler if there was an actual `Error` here

Reply:
As @juliusmarminge already pointed out, it'd be nice if it was something like

```ts
if (result.error) {
  throw result.error
}
```

tuples work well too

```ts
const [err, result] =  await parser['~validate'](value);

if (err) {
  throw err;
}
return result
```

Reply:
I think forcing schema libraries to use a specific error class to be compatible with Standard Schema is not a good idea, as it would unnecessarily compromise bundle size, performance (creating `Error` instances is expensive), and flexibility.